src/main/java/bio/terra/workspace/service/workspace/flight/CreateControlledResourceFlight.java

```diff
@@ -0,0 +1,57 @@
+package bio.terra.workspace.service.workspace.flight;
+
+import bio.terra.stairway.Flight;
+import bio.terra.stairway.FlightMap;
+import bio.terra.workspace.common.utils.FlightBeanBag;
+import bio.terra.workspace.service.datareference.flight.GenerateReferenceIdStep;
+import bio.terra.workspace.service.job.JobMapKeys;
+import bio.terra.workspace.service.resource.controlled.ControlledResource;
+import bio.terra.workspace.service.resource.controlled.gcp.CreateGcsBucketStep;
+
+/**
+ * Flight for creation of a controlled resource. Some steps are resource-type-agnostic, and others
+ * depend on the resource type. The latter must be passed in via the input parameters map with keys
+ */
+public class CreateControlledResourceFlight extends Flight {
+
+  public CreateControlledResourceFlight(FlightMap inputParameters, Object applicationContext) {
+    super(inputParameters, applicationContext);
+    final FlightBeanBag flightBeanBag = FlightBeanBag.getFromObject(applicationContext);
+
+    // Generate a new resource UUID
+    addStep(new GenerateControlledResourceIdStep());
+    // Genderate reference ID for the workspace_data_reference table
+    addStep(new GenerateReferenceIdStep());
+
+    // store the resource metadata in CloudSQL
```

Comment:
nit: this stores the resource metadata in WSM's underlying database, which happens to be CloudSQL in most environments. But that isn't necessarily be true, e.g. locally it's just a postgres DB.